### PR TITLE
Associate a Cluster Name to Home and Foreign Clusters

### PR DESCRIPTION
**Pull request by aleoli**
# Description

This PR adds a "nice" name for our cluster. This can be set in the `ClusterConfig` CR

The remote clusters that will discover us will found our name in ForeignCluster in `Spec.ClusterIdentity.ClusterName`

If the `ClusterName` is updated in the ClusterConfig, the `ForeignCluster` on the already peered remote clusters is not updated, but the future clusters will discover it with the new name.

### CODE REVIEW

File: api/discovery/v1alpha1/foreigncluster_types.go [hunk 1]
```diff
@@ -42,12 +42,17 @@ type ForeignClusterSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	ClusterID        string        `json:"clusterID"`
-	Namespace        string        `json:"namespace"`
-	Join             bool          `json:"join"`
-	ApiUrl           string        `json:"apiUrl"`
-	DiscoveryType    DiscoveryType `json:"discoveryType"`
-	AllowUntrustedCA bool          `json:"allowUntrustedCA"`
+	ClusterIdentity  *ClusterIdentity `json:"clusterIdentity"`
```
Comment (frisso) [inline api/discovery/v1alpha1/foreigncluster_types.go:45]: Why this change? 
`ClusterIdentity`  looks weird; `ClusterID` looks much more common (if it is an ID, of course).

File: api/discovery/v1alpha1/foreigncluster_types.go [hunk 2]
```diff
@@ -42,12 +42,17 @@ type ForeignClusterSpec struct {
...
+	Namespace        string           `json:"namespace"`
+	Join             bool             `json:"join"`
```
Comment (frisso) [inline api/discovery/v1alpha1/foreigncluster_types.go:47]: What does this mean?
`Join` looks meaningless to me. `AutoJoin`, maybe?
@palexster ?

Comment (aleoli) [reply api/discovery/v1alpha1/foreigncluster_types.go]: This is the flag to set to start the peering process, that can be set or because of an `AutoJoin` in Config or manually (or by dashboard) by the user

Comment (palexster) [reply api/discovery/v1alpha1/foreigncluster_types.go]: I totally agree that "Join" does not explicit well the status of the peering, maybe something like "Peered". I suggest to handle this in a dedicated PR, since I feel it requires a wider refactoring which involve many components.

**Review by palexster — APPROVED** [review]

**Review by nappozord — APPROVED** [review]
